core/core.js

```diff
@@ -65,18 +65,15 @@ function createMu (options) {
 
   function dispatch (message, cb) {
     var hash
-    var digest
 
     assert(message, 'dispatch requires a valid message')
     assert(cb && (typeof cb === 'function'), 'dispatch requries a valid callback handler')
     logger.debug('dispatching message: ' + message)
 
     if (!(message.pattern || message.response)) {
-      hash = crypto.createHash('sha256')
-      hash.update('' + cb)
-      digest = hash.digest('hex')
-      router.addRoute(null, {muid: digest, tf: cb, type: 'callback'})
-      router.route({pattern: message, protocol: {path: [digest], trace: [digest], ttl: DEFAULT_TTL}}, cb)
+      hash = md5('' + cb)
```

Comment:
I'm not sure though - I think what @pelger is trying to do is intrinsically associate the callback with the pattern - so if the *same* function (or, to be exact, the same function content) is registered as cb it has the same id 

not sure on the reason for this, we talked about it briefly at nodeconf but weren't able to conclude the convo